Instantiating Cleaner causes crash on Android 13

Related to element-hq/element-x-android#2346

When launching an Element X version that relies on uniffi-rs v0.26.0 or higher on a Pixel 4a running GrapheneOS based on Android 13, the app crashes with the following error:
```
java.lang.AssertionError: Erroneously trying to create Cleaner in zygote
```
The full crash log is available here: [crash.log](https://github.com/mozilla/uniffi-rs/files/14635727/crash.log)

Based on the AOSP commit history and GrapheneOS's commit history, this AssertionError was removed starting with Android 14 (API level 34), so it seems that using `AndroidSystemCleaner` is not supported on all Android OSes with API level 33, especially GrapheneOS.
#1996 updated the min API level for the new Cleaner to 33, it might need to be raised to 34.

--- a/uniffi_bindgen/src/bindings/kotlin/templates/ObjectCleanerHelperAndroid.kt
+++ b/uniffi_bindgen/src/bindings/kotlin/templates/ObjectCleanerHelperAndroid.kt
@@ -2,14 +2,15 @@
 {{- self.add_import("androidx.annotation.RequiresApi") }}
 
 private fun UniffiCleaner.Companion.create(): UniffiCleaner =
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
         AndroidSystemCleaner()
     } else {
         UniffiJnaCleaner()
     }
 
 // The SystemCleaner, available from API Level 33.
-@RequiresApi(Build.VERSION_CODES.TIRAMISU)
+// Some API Level 33 OSes do not support using it, so we require API Level 34.
+@RequiresApi(Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
 private class AndroidSystemCleaner : UniffiCleaner {
     val cleaner = android.system.SystemCleaner.cleaner()
 
@@ -17,7 +18,7 @@ private class AndroidSystemCleaner : UniffiCleaner {
         AndroidSystemCleanable(cleaner.register(value, cleanUpTask))
 }
 
-@RequiresApi(Build.VERSION_CODES.TIRAMISU)
+@RequiresApi(Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
 private class AndroidSystemCleanable(
     private val cleanable: java.lang.ref.Cleaner.Cleanable,
 ) : UniffiCleaner.Cleanable {

